src/main/java/org/snomed/snowstorm/ecl/validation/ECLEdgeCaseHandlerService.java

```diff
@@ -0,0 +1,131 @@
+import io.kaicode.elasticvc.domain.Commit;
+import org.snomed.langauges.ecl.ECLQueryBuilder;
+import org.snomed.langauges.ecl.domain.refinement.SubAttributeSet;
+import org.snomed.snowstorm.ecl.ECLQueryService;
+import org.snomed.snowstorm.ecl.domain.expressionconstraint.SExpressionConstraint;
+import org.snomed.snowstorm.ecl.domain.expressionconstraint.SRefinedExpressionConstraint;
+import org.snomed.snowstorm.ecl.domain.expressionconstraint.SSubExpressionConstraint;
+import org.snomed.snowstorm.ecl.domain.refinement.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.snomed.snowstorm.core.data.domain.Concepts.CONCEPT_MODEL_DATA_ATTRIBUTE;
+import static org.snomed.snowstorm.ecl.domain.expressionconstraint.SExpressionConstraintHelper.MISSING;
+
+@Service
+public class ECLEdgeCaseHandlerService implements CommitListener {
+
+	private static final String RETURN_ALL_CONCRETE_ATTRIBUTES_ECL_QUERY = "< " + CONCEPT_MODEL_DATA_ATTRIBUTE;
+
+	private static final List<String> CACHED_CONCRETE_CONCEPT_IDS = new ArrayList<>();
+
+	@Autowired
+	private ECLQueryBuilder eclQueryBuilder;
+
+	@Autowired
+	private ECLQueryService eclQueryService;
+
+	@Autowired
+	private VersionControlHelper versionControlHelper;
+
+	@Override
+	public final void preCommitCompletion(final Commit commit) throws IllegalStateException {
+		CACHED_CONCRETE_CONCEPT_IDS.clear();
+	}
+
+	public SExpressionConstraint replaceIncorrectConcreteAttributeValue(final String ecl, final String branch, final PageRequest pageRequest) {
+		SExpressionConstraint expressionConstraint = null;
+		if (ecl != null) {
+			if (CACHED_CONCRETE_CONCEPT_IDS.isEmpty()) {
+				cacheConcreteConceptIds(branch, pageRequest);
+			}
+			expressionConstraint = (SExpressionConstraint) eclQueryBuilder.createQuery(ecl);
+			if (expressionConstraint instanceof SRefinedExpressionConstraint) {
+				extractSEclAttributes((SRefinedExpressionConstraint) expressionConstraint).stream().filter(Objects::nonNull).filter(SEclAttribute::isConcreteValueQuery)
+						.forEach(sEclAttribute -> sEclAttribute.getConceptIds().stream().filter(concreteConceptId -> !CACHED_CONCRETE_CONCEPT_IDS.contains(concreteConceptId))
+								.forEach(concreteConceptId -> setPlaceholder(sEclAttribute)));
+			}
+		}
+		return expressionConstraint;
+	}
+
+	private void setPlaceholder(final SEclAttribute sEclAttribute) {
+		if (sEclAttribute.getNumericValue() != null) {
+			sEclAttribute.setNumericValue(String.valueOf(Float.MAX_VALUE));
+		} else if (sEclAttribute.getStringValue() != null) {
+			sEclAttribute.setStringValue(MISSING);
+		}
+	}
+
+	private List<SEclAttribute> extractSEclAttributes(final SRefinedExpressionConstraint expressionConstraint) {
```

Comment:
We will need to call a method like this recursively to cover all possible cases. The logic here may cover the unit tests but ECL can contain nested queries to any level. I can help with this if needed.